api/authenticator.py

```diff
@@ -1674,6 +1679,19 @@ class BasicAuthenticationProvider(AuthenticationProvider, HasSelfTests):
     TEST_PASSWORD_DESCRIPTION_REQUIRED = _("The password for the Test Identifier.")
     TEST_PASSWORD_DESCRIPTION_OPTIONAL = _("The password for the Test Identifier (above, in previous section).")
 
+    LIBRARY_SETTINGS = [
+        { "key": HTTP_BASIC_OAUTH_ENABLED,
+          "label": _("Enable OAuth for HTTP Basic Auth"),
+          "description": _("Enable authentication with bearer tokens generated via basic auth credentials"),
+          "type": "select",
+          "options": [
+              { "key": "false", "label": _("Disabled") },
+              { "key": "true", "label": _("Enabled") },
+          ],
+          "default": "false",
+        },
+    ] + AuthenticationProvider.LIBRARY_SETTINGS
```

Comment:
I added this setting to get tests to pass (`tests/admin/controller/test_patron_auth.py`), but noticed that it is causing them to also show up in the Library Settings on the Patron Authentication page, is this fine or is there something else I should be doing?

Reply:
This looks like it's the correct behavior. I checked out the `develop` branch and added a basic authenticator and looked at the settings and these were there sans the new OAuth setting. Marking as resolved.